src/client/datascience/kernel-launcher/kernelLauncher.ts

```diff
@@ -119,10 +123,11 @@ export class KernelLauncher implements IKernelLauncher {
         return ports;
     }
 
-    private async getKernelConnection(): Promise<IKernelConnection> {
+    private async getKernelConnection(
+        kernelConnectionMetadata: KernelSpecConnectionMetadata | PythonKernelConnectionMetadata
+    ): Promise<IKernelConnection> {
         const ports = await this.getConnectionPorts();
         return {
-            version: 1,
```

Comment:
This change was crucial.
The kernels would crash when this version is `1`.
Looking at jupyter source & generated configs (at least latest version), they don't set a version number.

Reply:
That's odd. I feel like that had been 1 for a while. I think since pulling in initial raw kernel stuff. In any case change seems fine.